packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/modules.ts

```diff
@@ -1,61 +1,80 @@
 /* eslint-disable import/no-extraneous-dependencies */
 import { ExternalModule, Type } from '@cdklabs/typewriter';
+import { CallableExpr } from './callable-expr';
+
+/**
+ * A class representing an external module that can be imported into a target module.
+ */
+export abstract class ImportableModule extends ExternalModule {
+  /**
+   * The name to import all targets in the module as.
+   */
+  public abstract importAs: string;
+}
+
+class PathModule extends ImportableModule {
+  public readonly join = CallableExpr.fromName(this, 'join');
+
+  public readonly importAs = 'path';
 
-class PathModule extends ExternalModule {
   public constructor() {
     super('path');
   }
 }
 
-class ConstructsModule extends ExternalModule {
+class ConstructsModule extends ImportableModule {
   public readonly Construct = Type.fromName(this, 'Construct');
 
+  public readonly importAs = 'constructs';
+
   public constructor() {
     super('constructs');
   }
 }
 
-class CoreModule extends ExternalModule {
+class CoreModule extends ImportableModule {
   public readonly Stack = Type.fromName(this, 'Stack');
   public readonly CustomResourceProviderBase = Type.fromName(this, 'CustomResourceProviderBase');
   public readonly CustomResourceProviderOptions = Type.fromName(this, 'CustomResourceProviderOptions');
 
+  public readonly determineLatestNodeRuntimeName = CallableExpr.fromName(this, 'determineLatestNodeRuntimeName');
+
+  public readonly importAs = 'cdk';
+
   public constructor() {
     super('../../../core');
   }
 }
 
-class CoreInternalStack extends ExternalModule {
-  public readonly Stack = Type.fromName(this, 'Stack');
+class LambdaModule extends ImportableModule {
+  public readonly Function = Type.fromName(this, 'Function');
+  public readonly SingletonFunction = Type.fromName(this, 'SingletonFunction');
+  public readonly FunctionOptions = Type.fromName(this, 'FunctionOptions');
+  public readonly Runtime = Type.fromName(this, 'Runtime');
+  public readonly RuntimeFamily = Type.fromName(this, 'RuntimeFamily');
+  public readonly Code = Type.fromName(this, 'Code');
 
-  public constructor() {
-    super('../../stack');
-  }
-}
+  public readonly determineLatestNodeRuntime = CallableExpr.fromName(this, 'determineLatestNodeRuntime');
 
-class CoreInternalCustomResourceProvider extends ExternalModule {
-  public readonly CustomResourceProviderBase = Type.fromName(this, 'CustomResourceProviderBase');
-  public readonly CustomResourceProviderOptions = Type.fromName(this, 'CustomResourceProviderOptions');
+  public readonly importAs = 'lambda';
 
   public constructor() {
-    super('../../custom-resource-provider');
+    super('../../../aws-lambda');
```

Comment:
All the hardcoded paths continue to worry me.

Reply:
Agree with the ones used in the `classes.ts`, but these are just used to initialize the `fqn` property in the super class for `ExternalModules`. This would then be used as `this.fqn` for the specific module. That said, I think paths used to initialize the super are okay. I moved everything else into a constant.